src/concurrency.md

```diff
@@ -232,6 +233,49 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[threadpool-walk]: #threadpool-walk
+<a name="threadpool-walk"></a>
+## Use a threadpool to calculate MD5 to all files in a folder
+
+[![threadpool-badge]][threadpool] [![num_cpus-badge]][num_cpus] [![walkdir-badge]][walkdir] [![cat-concurrency-badge]][cat-concurrency][![cat-filesystem-badge]][cat-filesystem]
+
+This example uses `threadpool`, `walkdir` and `md5` crates to calculate the MD5 for every file present in the current directory. A threadpool `pool` is created using the number of cpus present in the system. Then every `entry` returned by `walkdir` is passed into this pool to perform the operations of reading and compute of md5. At the end the program waits for all jobs to finish.
+
+```rust,no_run
+extern crate walkdir;
+extern crate md5;
+extern crate num_cpus;
+extern crate threadpool;
+
+use walkdir::WalkDir;
+use std::fs::File;
+use std::io::BufReader;
+use std::io::prelude::*;
+use threadpool::ThreadPool;
+
+fn main() {
+    let cpus = num_cpus::get();
+    let pool = ThreadPool::new(cpus);
+
+    for entry in WalkDir::new(".")
```

Comment:
lets go wild and compute the sha1 of all *.iso files in given directory or all files listed in 
or the ones listed in 
https://cloud-images.ubuntu.com/trusty/current/SHA1SUMS
https://cloud-images.ubuntu.com/trusty/current/

In this way we could have an actual useful example. something along these lines - "Verify in parallel downloaded  files consistency with SHA1 algorithm"